app/models/user.rb

```diff
@@ -0,0 +1,41 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :bigint           not null, primary key
+#  admin                  :boolean          default(FALSE), not null
+#  email                  :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  name                   :string
+#  remember_created_at    :datetime
+#  reset_password_sent_at :datetime
+#  reset_password_token   :string
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+# Indexes
+#
+#  index_users_on_email                 (email) UNIQUE
+#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
+#
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:authsch]
+
+  validates :name, presence: true
+
+  def self.from_omniauth(access_token)
```

Comment:
You can copy the bit better coded example from [https://github.com/kir-dev/airsoft/blob/master/app/models/user.rb](https://github.com/kir-dev/airsoft/blob/master/app/models/user.rb).

What is important is that the name comes in  the `.extra.raw_info` object